### DISCUSSION TITLE
How to pass dataframe or yaml object to quarto through cli ?

### DISCUSSION THREAD

**Discussion by sunvitD** [quarto-dev/quarto-cli#11689] [Q&A] [answered]
### Description

I am trying to pass a `dataframe` or `yaml object` to `Quarto` through `quarto cli` but I am getting this error: `ERROR: YAML file - not found.`

I have tried passing string as I cannot share a `.yml file` since writing file is not possible in my situation so is there any alternative ?  

I have tried below code but it doesn't seem to work:
```
import polars as pl
import yaml

from pandas import json_normalize
from yaml import SafeLoader, load

df = pl.DataFrame(
    {"foo": [1, 2, 3],
        "bar": [6.0, 7.0, 8.0],
        "ham": ["a", "b", "c"],})

df_pd = df.to_pandas()
df_yaml = yaml.dump(df_pd.to_dict(orient='records'),default_flow_style=None)

cmd_str = f"quarto render 2-test.qmd --execute-params {df_yaml} --output test_16-12-2024.html"
cmd_str
``` 
`quarto render 2-test.qmd --execute-params - {bar: 6.0, foo: 1, ham: a} - {bar: 7.0, foo: 2, ham: b} - {bar: 8.0, foo: 3, ham: c} --output test_16-12-2024.html`

Would appreciate any help !!

**Comment by mcanouil** [comment]
I don't think you are asking the right question here.
Before talking about the means, what's the end goal?

**Reply by sunvitD** [reply]
well I was running `quarto` as a `subprocess` in python script and I needed to pass `dataframe object` from the py script that was running this subprocess to the quarto. Writing a file `csv` or `yaml` to disk was not possible for some reason so I wanted to understand if there is any other way to pass `dataframe` as `parameters` to the quarto through cli ?

I tried converting `yaml` object to `string` and patched it with `--execute-params` in the command but that didn't work.

This is finally sorted as I am able  to write files in the working directory containing both py scripts & quarto so reading the dataframe directly by quarto is now possible for me but still it would be good to know  if passing `yaml` as `string` is possible in qurto render command ?

**Reply by mcanouil** [reply, marked as answer]
> well I was running quarto as a subprocess in python script and I needed to pass dataframe object from the py script that was running this subprocess to the quarto. Writing a file csv or yaml to disk was not possible for some reason so I wanted to understand if there is any other way to pass dataframe as parameters to the quarto through cli ?

To clarify, Quarto cannot have access to your environment (only exported environment variables) as any CLI tools running in `subprocess`.

> I tried converting yaml object to string and patched it with --execute-params in the command but that didn't work.

I don't understand what you tried here. `--execute-params` takes a YAML file. `--execute-param` takes a `key:value` string.
See `quarto render --help`.

As long as you respect the YAML format, you can do whatever you want in both flags.
For `--execute-param`, `value` is a scalar, you can't have a nested YAML string, for complex YAML values, use `--execute-params`.

In any cases, you'll have to post-process the parameter to give back the value its dataframe shape.

**Reply by sunvitD** [reply]
Thanks for the quick response. Its much more clear now. Appreciate your help !!!
